src/Traits/MessageResponseTrait.php

```
@@ -29,6 +29,9 @@ public function getProtocolVersion(): string
         return $this->response->getProtocolVersion();
     }
 
+    /**
+     * @return MessageInterface
```

Nit: This PHPDoc is redundant since it's already expressed by the native return type

---

I agree that this is redundant but if I remove it I get the following error from PHPStan.
```
$ composer run-script phpstan
> phpstan analyse --no-progress --memory-limit 256M
Note: Using configuration file /.../Git/elasticsearch-php/phpstan.neon.
 ------ ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- 
  Line   Traits/MessageResponseTrait.php (in context of class Elastic\Elasticsearch\Response\Elasticsearch)                                                                                             
 ------ ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- 
  :34    Method Elastic\Elasticsearch\Response\Elasticsearch::withProtocolVersion() should return static(Elastic\Elasticsearch\Response\Elasticsearch) but returns Psr\Http\Message\ResponseInterface.  
         🪪  return.type                                                                                                                                                                                
 ------ ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- 
```
I'm trying to understand where is the issue here, any idea? Thanks.

---

This looks to be because Elasticsearch is actually violating the ResponseInterface interface by returning an instance of the underlying ResponseInterface rather then the object it's being called on.

https://github.com/php-fig/http-message/blob/402d35bcb92c70c026d1a6a9883f06b2ead23d71/src/MessageInterface.php#L39

By adding the PHPDoc to the trait it's overwriting the one from the ResponseInterface which is why it silences the error.

I have opened a PR to fix this properly: https://github.com/elastic/elasticsearch-php/pull/1440

---

Thanks @AJenbo nice catch! I'll ask to wait the #1440 until we merge this PR: Moreover, can you rebase the PR to 8.17 branch? Thanks.

---

Done, I also set it to target the 8.17 branch so you can make it part of this PR.